packages/@aws-cdk/aws-iam/lib/grouping-by-actions-principal.ts

```diff
@@ -0,0 +1,68 @@
+import * as crypto from 'crypto';
+import { IIdentity } from './identity-base';
+import { PolicyStatement } from './policy-statement';
+import { AddToPrincipalPolicyResult, IGrantable, IPrincipal, PrincipalPolicyFragment } from './principals';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * An IAM Principal that wraps a different Principal,
+ * and groups multiple Policy Statements that share the same 'actions' property together,
+ * thus compressing the overall size of the Policy attached the given Principal,
+ * and making it less likely it hits the 10240 bytes IAM limit.
+ */
+export class GroupingByActionsPrincipal extends Construct implements IPrincipal, IGrantable {
+  public readonly assumeRoleAction: string;
+  public readonly policyFragment: PrincipalPolicyFragment;
+  public readonly principalAccount?: string | undefined;
+  public readonly grantPrincipal: IPrincipal;
+
+  /** The original principal that this class wraps. */
+  public readonly wrappedIdentity: IIdentity;
+  private readonly statements: { [key: string]: { statement: PolicyStatement, grantResult: AddToPrincipalPolicyResult } };
+
+  constructor(grantPrincipal: IIdentity, id: string) {
+    super(grantPrincipal as unknown as Construct, id);
```

Comment:
This cast looks icky. Why this?

Does `GroupingByActionsPrincipal` need to be a `Construct` in the first place?